new-nextjs-client/src/components/index-page/CourseSearchWindow/SearchControl/SearchControl.tsx

```diff
@@ -105,6 +105,18 @@ function SearchControl(props: Props) {
     }
   }, [selectValue]);
 
+  useEffect(() => {
+    if (selectValue) {
+      getAllDepartmentCourses(selectValue).then((courses) =>
```

Comment:
We do not want to make BE call every time the input changes. Use the existing `searchResults` state for filtering.